modules/ducktests/tests/checks/utils/check_marks.py

```diff
@@ -0,0 +1,240 @@
+            args = ctx.injected_args
+            assert len(args) == 4
+            assert ctx.function() == (args['pair_1'], args['pair_2'], args['i'], args['j'])
+
+        assert len(list(filter(lambda x: x.function_name == function_ignore.__name__, context_list))) == 2
+        assert len(list(filter(lambda x: x.ignore, context_list))) == 1
+
+    @staticmethod
+    def _check_injection(context_list, *, versions, global_args=None, pairs=False):
+        if global_args:
+            global_versions = global_args['ignite_versions']
+
+            if isinstance(global_versions, str):
+                assert not pairs
+
+                check_versions = [IgniteVersion(global_versions)]
+            elif isinstance(global_args['ignite_versions'], tuple):
+                assert pairs
+
+                check_versions = [tuple(map(IgniteVersion, global_versions))]
+            elif pairs:
+                check_versions = list(map(lambda x: (IgniteVersion(x[0]), IgniteVersion(x[1])), global_versions))
+            else:
+                check_versions = list(map(IgniteVersion, global_versions))
+        else:
+            check_versions = list(map(IgniteVersion, versions)) if not pairs else \
+                list(map(lambda x: (IgniteVersion(x[0]), IgniteVersion(x[1])), versions))
+
+        assert len(context_list) == len(check_versions)
+
+        for i, ctx in enumerate(sorted(context_list, key=lambda x: x.function())):
+            assert ctx.function() == check_versions[i]
+
+
+class CheckVersionIf:
+    """
+    Checks @version_if parametrization.
+    """
+    def check_common(self):  # pylint: disable=R0201
+        """
+        Check common scenarios with @ignite_versions parametrization.
+        """
+        @version_if(lambda ver: ver != V_2_8_0, variable_name='ver')
+        @ignite_versions(str(DEV_BRANCH), str(V_2_8_0), version_prefix='ver')
+        def function_1(ver):
+            return IgniteVersion(ver)
+
+        @version_if(lambda ver: ver > V_2_7_6, variable_name='ver_1')
+        @version_if(lambda ver: ver < V_2_8_0, variable_name='ver_2')
+        @ignite_versions((str(V_2_8_1), str(V_2_8_0)), (str(V_2_8_0), str(V_2_7_6)), version_prefix='ver')
+        def function_2(ver_1, ver_2):
+            return IgniteVersion(ver_1), IgniteVersion(ver_2)
+
+        @ignite_versions(str(DEV_BRANCH), str(V_2_8_0))
+        def function_3(ignite_version):
+            return IgniteVersion(ignite_version)
+
+        context_list = expand_function(func=function_1, sess_ctx=mock_session_ctx())
+        context_list += expand_function(func=function_2, sess_ctx=mock_session_ctx())
+        context_list += expand_function(func=function_3, sess_ctx=mock_session_ctx())
+
+        assert len(context_list) == 6
+
+        assert next(filter(lambda x: x.injected_args['ver'] == str(V_2_8_0), context_list)).ignore
+        assert not next(filter(lambda x: x.injected_args['ver'] == str(DEV_BRANCH), context_list)).ignore
```

Comment:
Why do you check only 'ver' and not others (ver_1, ver_2, ignit_verson)?

Reply:
Different options are tested in other checks, see below